From 445a5255265b40d4964d37874efb00d0f8981e5d Mon Sep 17 00:00:00 2001
From: Robert Marshall <rmarshall@gitlab.com>
Date: Thu, 13 Nov 2025 00:11:46 -0500
Subject: [PATCH] Patch curl OpenSSL m4 for AL2-FIPS pkg-config

- Updates the OpenSSL m4 configuration to point at the `openssl11`
  pkg-config file. This eliminates accidental linkage to libraries other
  than OpenSSL 1.1.x in FIPS mode.

Signed-off-by: Robert Marshall <rmarshall@gitlab.com>
---
 m4/curl-openssl.m4 | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/m4/curl-openssl.m4 b/m4/curl-openssl.m4
index 20e3284189..81ad598a18 100644
--- a/m4/curl-openssl.m4
+++ b/m4/curl-openssl.m4
@@ -79,7 +79,7 @@ if test "x$OPT_OPENSSL" != xno; then
       dnl specify PKG_CONFIG_LIBDIR we're only looking where
       dnl the user told us to look
       OPENSSL_PCDIR="$OPT_OPENSSL/lib/pkgconfig"
-      if test -f "$OPENSSL_PCDIR/openssl.pc"; then
+      if test -f "$OPENSSL_PCDIR/openssl11.pc"; then
         AC_MSG_NOTICE([PKG_CONFIG_LIBDIR will be set to "$OPENSSL_PCDIR"])
         PKGTEST="yes"
       fi
@@ -87,7 +87,7 @@ if test "x$OPT_OPENSSL" != xno; then
       if test "$PKGTEST" != "yes"; then
         # try lib64 instead
         OPENSSL_PCDIR="$OPT_OPENSSL/lib64/pkgconfig"
-        if test -f "$OPENSSL_PCDIR/openssl.pc"; then
+        if test -f "$OPENSSL_PCDIR/openssl11.pc"; then
           AC_MSG_NOTICE([PKG_CONFIG_LIBDIR will be set to "$OPENSSL_PCDIR"])
           PKGTEST="yes"
         fi
@@ -341,7 +341,7 @@ if test "x$OPT_OPENSSL" != xno; then
       fi
     fi
     check_for_ca_bundle=1
-    LIBCURL_PC_REQUIRES_PRIVATE="$LIBCURL_PC_REQUIRES_PRIVATE openssl"
+    LIBCURL_PC_REQUIRES_PRIVATE="$LIBCURL_PC_REQUIRES_PRIVATE openssl11"
   fi
 
   test -z "$ssl_msg" || ssl_backends="${ssl_backends:+$ssl_backends, }$ssl_msg"
-- 
2.51.2

